exchanges/stream/websocket_types.go

```diff
@@ -25,13 +23,21 @@ const (
 
 type subscriptionMap map[any]*subscription.Subscription
 
+type state int
+
+const (
+	uninitialised state = iota
+	disconnected
+	connecting
+	connected
+)
+
 // Websocket defines a return type for websocket connections via the interface
 // wrapper for routine processing
 type Websocket struct {
 	canUseAuthenticatedEndpoints bool
 	enabled                      bool
-	Init                         bool
-	connected                    bool
+	state                        state
 	connecting                   bool
```

Comment:
connecting field in this struct can be removed

Reply:
Absolutely. Thought it was removed. Thanks for catching.

Fixed ac5fe67b